File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if not head:
            return 
        values = []
        while head:            
            values.append(head.val)
            head = head.next

        counts = Counter(values)
        for key in Counter(values):
            if counts[key] != 1:
                del counts[key]

        result_head = ListNode(-101)
        result = result_head
        for value in sorted(counts):
            result.next = ListNode(value) 
            result = result.next           
        return result_head.next  

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;
        ListNode dummy = new ListNode(0);
        ListNode node = dummy;
        while (head != null) {
            if ((head.next != null && head.val != head.next.val) || head.next == null) {
                node.next = new ListNode(head.val);
                node = node.next;
                
            } else {
                while (head.next != null && head.val == head.next.val) {
                    head = head.next;
                }
            }
            head = head.next;
        }
        return dummy.next;
    }
}